### ISSUE TITLE
install fails if node version >20.x.x even if it claims it supports up to 22.x.x (current lts)

### ISSUE THREAD

**Issue by jacopomazzoni** [strapi/pack-up#24]
### Required System information

- Node.js version: 20.18.1
- NPM version: 10.8.2
- Strapi version: 5.5.1
- Database: sqlite
- Operating system: macos
- Is your project Javascript or Typescript: TS

### Describe the bug
installing using yarn as described in the homepage throws an error on the latest node 24 that strapi supports versions 18-22.
switching to 22 allows to get to the install wizard up to a certain point when it throws this error:
error @strapi/pack-up@5.0.2: The engine "node" is incompatible with this module. Expected version ">=18.0.0 <=20.x.x". Got "22.12.0"

then you need to downgrade to node 20, then it works,
suboptimal experience, is there a valid reason why @strapi/pack-up@5.0.2 wants node version 20 instead of 22 or could this requirement be removed? or made clearer at the first check? 

### Steps to reproduce the behavior

1. install node 22
2. run yarn create strapi
3. See error

### Expected behavior

 either require node 22 or 20, make up your mind

### Screenshots

na
### Code snippets

na
### Additional context

na

**Comment by thanhnd1o2** [comment]
I got the same issue

**Comment by coditive-przemek-h** [comment]
The same issue here. Every app created with official docs (`npx create-strapi@latest my-strapi-project`) generates this error.

**Comment by Deanfei** [comment]
Same here. I am wondering when there will be a new release for this pack-up. It looks the fixing is merged but not released yet. Unfortunately I couldn't install node 20 on my manchine for specific security reasons. I have to stick with node 22 and it's currently urgently blocking. I have to switch to Payload CMS as an alternative for now but still prefer to try strapi as my first choice.

**Comment by sisirkon** [comment]
Hello @derrickmehaffy ,

The comment below mentioned a release a month ago since the MR was merged, is there a way to to track it, this would be helpful to avoid restricting usage of specific package version and use the latest.

https://github.com/strapi/pack-up/issues/22#issuecomment-2539373278

**Comment by Deanfei** [comment]
Yes, please. I am currently blocked by the node22 restrictions. It'll be really appreciated if this can be resolved ASAP.

**Comment by hynding** [comment]
I was running into the same issue so I gave up using yarn as my package manager and switched to `pnpm` instead. Everything appears to be working just fine after deleting `node_modules` and `yarn.lock` then reinstalling with `pnpm install`

**Comment by sisirkon** [comment]
@hynding curious what version of node are you using , because the tagged version 5.0.2 that is linked to most of the strapi starter project is explicitly calling the node version to be 20 or below  https://github.com/strapi/pack-up/blob/%40strapi/pack-up%405.0.2/package.json#L116. If 20 or less , I am guessing backward compatibility for the types module.

**Comment by hynding** [comment]
@sisirkon I'm using the latest LTS as of today (v22.13.0) and recently upgrade to Strapi 5.0.7 but initially started with 5.0.0, I believe (via starter project using `npx`). I looked into what was pulled into `node_modules` when using `yarn` and saw the `package.json` file for `@strapi/pack-up` was still max targeting v20 which led me to believe the NPM repo was still out of date (thought the package.json version was `5.0.2`) hence Yarn's cache was also not updated. The Strapi projects, however, all appeared to be using `pnpm` so I figured I'd give that a try instead and, voila, it worked without issue.

I'll update if it turns out I have deployment issues but all the major cloud deployment providers appear to support `pnpm` when its lock file is detected so I'm keeping my fingers crossed everything will run smoothly over to production.

**Comment by sisirkon** [comment]
@hynding Seems like npm , yarn , pnpm deal with dependencies differently by default.

pnpm unless the config set pnpm seems to ignore engine of dependencies, only respects engine the parent by default https://pnpm.io/npmrc#engine-strict

npm and yarn seem to have variations of the same issue but with different side effects , what a mess 🤦  across packagers.

Hopefully, it works for you 🤞

**Comment by Deanfei** [comment]
So funny for this issue.

**Comment by hramos-ps** [comment]
> [@sisirkon](https://github.com/sisirkon) I'm using the latest LTS as of today (v22.13.0) and recently upgrade to Strapi 5.0.7 but initially started with 5.0.0, I believe (via starter project using `npx`). I looked into what was pulled into `node_modules` when using `yarn` and saw the `package.json` file for `@strapi/pack-up` was still max targeting v20 which led me to believe the NPM repo was still out of date (though the package.json version was `5.0.2`) hence Yarn's cache was also not updated. The Strapi projects, however, all appeared to be using `pnpm` so I figured I'd give that a try instead and, voila, it worked without issue.
> 
> I'll update if it turns out I have deployment issues but all the major cloud deployment providers appear to support `pnpm` when its lock file is detected so I'm keeping my fingers crossed everything will run smoothly over to production.

I came with this same problem, started a project with fresh npx install and package.json was generated with `"node": ">=18.0.0 <=22.x.x"` but when I tried to use node 22 came into `@strapi/pack-up` compatibility problem.

```
npm error engine Not compatible with your version of node/npm: @strapi/pack-up@5.0.2
npm error notsup Not compatible with your version of node/npm: @strapi/pack-up@5.0.2
```

So I downgraded to node 20 for now.

**Comment by adminy** [comment]
any plans on pushing master fix to npm?

**Comment by derrickmehaffy** [comment]
You'll need to update your Strapi version as we updated packup in later versions.

If you use something like Strapi 5.8.0 (the latest) you shouldn't hit any issues (make sure you delete your lock file when you upgrade)

**Comment by derrickmehaffy** [comment]
The version of pack-up with node 22 support is 5.0.2

https://www.npmjs.com/package/@strapi/pack-up/v/5.0.2

**Comment by adriankirchner** [comment]
> The version of pack-up with node 22 support is 5.0.2
> 
> https://www.npmjs.com/package/@strapi/pack-up/v/5.0.2

5.0.2 still defines engine requirement <= 20: https://github.com/strapi/pack-up/blob/%40strapi/pack-up%405.0.2/package.json#L116

This version is used in the latest Strapi Release 5.8.1

**Comment by sisirkon** [comment]
> You'll need to update your Strapi version as we updated packup in later versions.
> 
> If you use something like Strapi 5.8.0 (the latest) you shouldn't hit any issues (make sure you delete your lock file when you upgrade)

Hello @derrickmehaffy ,

Just for reference, I have 5.9.0, will lock file and node modules deleted.

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@strapi/pack-up@5.0.2',
npm WARN EBADENGINE   required: { npm: '>=6.0.0', node: '>=18.0.0 <=20.x.x' },
npm WARN EBADENGINE   current: { node: 'v21.7.1', npm: '10.5.0' }
npm WARN EBADENGINE }


From the lock file:

    **"node_modules/@strapi/core": {
      "version": "5.9.0",**
      "resolved": "*********************",
      "integrity": "******************,
      "dependencies": {
        "@koa/cors": "5.0.0",
        "@koa/router": "12.0.2",
        "@paralleldrive/cuid2": "2.2.2",
        "@strapi/admin": "5.9.0",
        "@strapi/database": "5.9.0",
        "@strapi/generators": "5.9.0",
        "@strapi/logger": "5.9.0",
        **"@strapi/pack-up": "5.0.2",**


I do agree packup was updated based on the commit https://github.com/strapi/pack-up/commit/4700b27c6f5c0fc9312bc473f03ad518d050d455 but a tag was never created.

**Comment by kimjansheden** [comment]
Installed strapi v5.9.0 today following the guide:
`npx create-strapi-app@latest cms --quickstart`

Get this error:
```
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: '@strapi/pack-up@5.0.2',
npm warn EBADENGINE   required: { npm: '>=6.0.0', node: '>=18.0.0 <=20.x.x' },
npm warn EBADENGINE   current: { node: 'v22.14.0', npm: '10.9.2' }
npm warn EBADENGINE }
```

Is this going to be fixed?

**Comment by jhoward1994** [comment]
Hi please try the latest version of this project [@strapi/pack-up@5.1.0](https://github.com/strapi/pack-up/releases/tag/%40strapi%2Fpack-up%405.1.0)

This should fix your issue, we can reopen it if it's not the case :)

**Comment by kimjansheden** [comment]
@jhoward1994 Thank you I will try it 😀

**Comment by joeldow** [comment]
I am getting this error again with 5.12.1.  What's the fix?
